wagtail/tests/utils.py

```diff
@@ -104,14 +104,14 @@ def assertLogs(self, logger=None, level=None):
         return _AssertLogsContext(self, logger, level)
 
     @contextmanager
-    def register_hook(self, hook_name, fn):
+    def register_hook(self, hook_name, fn, order=0):
         from wagtail.wagtailcore import hooks
 
-        hooks.register(hook_name, fn)
+        hooks.register(hook_name, fn, order)
         try:
             yield
         finally:
-            hooks.get_hooks(hook_name).remove(fn)
+            hooks._hooks[hook_name].remove((fn, order))
```

Comment:
No, definitely not. As an app author, I define a hook for a reason that's internal to my app; I don't want other apps unilaterally deciding that my hook isn't necessary.

If there's a legitimate need to disable functionality in (say) wagtailimages, then by all means we can make that configurable _within wagtailimages_ - that way, we don't have other apps forcing configuration changes that wagtailimages wasn't designed to handle.